packages/fx-core/src/core/middleware/projectMigratorV3.ts

```diff
@@ -406,6 +410,31 @@ export async function manifestsMigration(context: MigrationContext): Promise<voi
     );
   }
 
+  const oldManifestPath = path.join(oldAppPackageFolderPath, MANIFEST_TEMPLATE_CONSOLIDATE);
+  const oldManifestExists = await fs.pathExists(path.join(context.projectPath, oldManifestPath));
+  if (!oldManifestExists) {
+    // templates/appPackage/manifest.template.json does not exist
+    throw MigrationError(
+      new Error(getLocalizedString("core.migrationV3.manifestNotExist")),
+      errorNames.manifestTemplateNotExist,
+      learnMoreLink
+    );
+  } else {
+    const res = await manifestUtils._readAppManifest(
+      path.join(context.projectPath, oldManifestPath)
+    );
+    if (res.isErr()) {
+      throw MigrationError(
+        new Error("templates/appPackage/manifest.template.json is invalid"),
+        errorNames.manifestTemplateInvalid
+      );
+    }
+    const teamsAppManifest = res.value;
+    if (teamsAppManifest.validDomains?.includes(validDomain.botWithValid)) {
```

Comment:
seems `v3MigrationUtils` just adds `validDomain.bot`. When will `validDomain.botWithValid` be added?

Reply:
validDomain.botWithValid is used to control debug task.json output key name.
Currently, we found no project missed validDomain.botWithValid. So there's no need to add it.

Reply:
Can you add a P2 task to add `validDomain.botWithValid`? The priority is not high since no template missed this placeholder. But since users may modify the template and we have hidden logic to fix ths missing value in V4, it's better to fix the template during migration.

Reply:
sure. Add it in issue
https://dev.azure.com/msazure/Microsoft%20Teams%20Extensibility/_workitems/edit/19286648